dlibs/dlib_userupdate.py

```diff
@@ -39,10 +38,53 @@ async def on_member_update(before, after, **kargs):
             pass
 
 
+async def send_joinlog(member, client, embed):
+    with db_hlapi(member.guild.id) as db:
+        if joinlog := db.grab_config("join-log"):
+            if logging_channel := client.get_channel(int(joinlog)):
+                try:
+                    logging_channel.send(embed=embed)
+                except discord.errors.Forbidden:
+                    pass
+
+
+def parsedate(indata):
+    return f"{time.strftime('%a, %d %b %Y %H:%M:%S', time.localtime(datetime.timestamp(indata)))} ({(datetime.utcnow() - indata).days} days ago)"
+
+
+async def on_member_join(member, **kargs):
+
+    inc_statistics([member.guild.id, "on-member-join", kargs["kernel_ramfs"]])
+
+    embed = discord.Embed(title=f"{member} joined.", description=f"*{member.mention} joined the server.*", color=0x758cff)
```

Comment:
shouldn't you check if join log is on before generating the embed? this goes for leave log as well